### DISCUSSION TITLE
"IF" Function in Methods (infolist)

### DISCUSSION THREAD

**Discussion by LucaCVG** [filamentphp/filament#10175] [Help] [answered]
Hi! What do you think is wrong with this portion of code? (in "color" method)
I'm comparing two (float) field values ​​from the same record, with a function, but the function always returns the string "danger".

ValueA = 10.000,00
ValueB = 15.000,00

` Components\TextEntry::make('valueA')
->color(function ($state): string {
  if (floatval($state) > floatval('valueB)) {
     return 'danger';
     } else {
     return 'success';
                }
     }),
]),`

I also tried this way, but the result doesn't change. The string "danger" is always returned:

`Components\TextEntry::make('valueA')
->color(function (): string {
  if (floatval('valueA') > floatval('valueB)) {
     return 'danger';
     } else {
     return 'success';
                }
     }),
]),`

**Comment by leandrocfe** [comment, marked as answer]
use `$record`

```php
TextEntry::make('valueA')
    ->color(fn ($record): string => $record->valueA > $record->valueB ? 'danger' : 'success')
```

**Reply by LucaCVG** [reply]
I'm about to cry :) it works! I tried for two days straight. 
Being a beginner is not nice at all! Thank you very much!

**Reply by leandrocfe** [reply]
No worries ;)
